app/controllers/admin/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Admin
+  class CommentsController < Admin::BaseController
+    load_and_authorize_resource
+
+    def index
+      @comments = grouped_comments(accessible_ordered_comments)
+      @unread_comments = grouped_comments(accessible_ordered_comments.find_since_last_login(current_user))
+      @posted_comments = grouped_comments(accessible_ordered_comments.find_comments_by_user(current_user))
+    end
+
+    private
+
+    def accessible_ordered_comments
+      Comment.accessible_by(current_ability).joins('INNER JOIN events ON commentable_id = events.id').order('events.title', 'comments.created_at DESC')
+    end
+
+    def grouped_comments(remarks)
```

Comment:
I think that there is no point in having the private functions.

We can just split the code of **index** into separate lines.

I also don't like the manual INNER JOIN. @raluka didn't we end up finding a solution for this that did not require this manual inner join? I hope you have notes of your previous code :)